Source/WTF/wtf/darwin/OSLogPrintStream.mm

```diff
@@ -58,12 +58,12 @@
     va_list backup;
     va_copy(backup, argList);
 ALLOW_NONLITERAL_FORMAT_BEGIN
-    size_t bytesWritten = vsnprintf(m_string.mutableData() + offset, freeBytes, format, argList);
+    size_t bytesWritten = vsnprintf(m_string.mutableSpanIncludingNullTerminator().subspan(offset).data(), freeBytes, format, argList);
```

Comment:
Too bad this will not detect if `offset` is past the null character.

Reply:
Does `subspan` (with release asserts) not check the bounds? It's UB to create an out of bounds `subspan` so crashing would be valid. Maybe we could request this?

Reply:
Right, I don't understand the concern here. `subspan()` would RELEASE_ASSERT if `offset` was out of bounds.

Reply:
A pointer to the end of the span, past the null character, isn't out of bounds as far as the `subspan` function knows. It's in the span but we need a position that is before the last byte in the span.

Reply:
Ah, yeah, `subspan(size())` would be a valid span but the `data()` would be invalid to dereference. It seems like it might be ok perf wise to do `RELEASE_ASSERT(!empty());` in `data()` since most(?) of the time you want the data you're about to loop over it anyway.

Reply:
I realized that for this case where we are passing the length remaining in the buffer separately:
- The length in the span does us little good because it's only checking the start of the area that vnsprintf is writing into. We are trusting our length computation.
- There's no reason we couldn't use `muableSpan()` without the hull terminator instead and do the one character stricter range checking on the call to `subspan()`, but there's not much benefit to that either.

Reply:
FYI, I believe the full buffer hardening model, which bans memset, (not yet enforced in WebKit, but coming soon) catches this case. Specifically:

* the sprintf family of functions should warns just like memset does, since it indexes into a buffer without a bounds check

* therefore this code needs to call a vsnpritnfSpan wrapper

* if subspan() has a past the end data(), its length is zero, and vsnprintfSpan will either print zero bytes or crash, depending on our preference

We can raise this issue in our next sync with the clang team, to make sure we’re on the same page.